episodes/structure-schema.Rmd

```diff
@@ -0,0 +1,464 @@
+10. This XML example stores data in form of natural text (<i>point to the two sentences</i>) and a number (<i>highlight the "1"</i>). Each of these data elements is enclosed in a start and end end tag. As mentioned before, these tags can be defined arbitrarily, so we state, that the first sentence is the title, the second sentence is a description and the number one represents a word count. We can now introduce a data hierarchy, declaring that all these data elements describe one example. We could now look for the "title" of the "example" and retrieve the data stored between those tags. XML is software- and hardware-independent and still widely used as a data exchange format. Though XML is human-readable, enclosing the data between start and end tags is quite an effort and meddles with the clarity of this format. So we are leaving the realm of markup languages...
+11. to introduce another data exchange format, which is recommended as a standard from the W3 consortium: JSON. JSON is short for JavaScript Object notation. It is inspired by the JavaScript representation for data objects and bears all the advantages XML has shown - like lightweight data exchange, software- and hardware-independency, and hierarchical structure - with an increased human-readability. For the remainder of this lesson, we will learn how to read, write and standardize JSON object literals.
+
+::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::: instructor
+## Lecture: JSON step-by-step
+
+**Slide set**: <a href="../files/slideSet/JsonStepByStep_CarpentriesIncubator_ScientificMetadata.pdf" target="_blank" download>JsonStepByStep.pdf</a>
+
+**Time estimate**:
+
+1. Let's have a closer look on the structure and syntax of JSON object literals. 
+2. In this example we have a JSON object, which describes a person named John.
+3. Objects are enclosed in curly braces.
+4. Within these objects, data entities are stored in key-value-pairs.
+5. Keys are always of the data type "string."
+6. Data values can be given as one of the following data types. They can be given as...
+7. strings. 
+8. String values can contain any kind of characters and are characteristically enclosed in parentheses. Each string, no matter how long it is, will be interpreted as a single value.
+9. Numbers...
+10. can be either integers or floats.
+11. Booleans... 
+12. Boolean values in JSON has one of two possible values. It is either true or false, or 1 for true and 0 for false, respectively.
```

Comment:
```suggestion
12. Boolean values in JSON have one of two possible values. It is either true or false.
```

I would not tell people to use 0 for false and 1 for true, if we want them to write schemas later... because this kind of inconsistency is actually something we want to avoid. That many places will for convenience sloppily auto-convert things into "truthy" and "falsy" values looks nice, but can be a source of possible errors and make data cleaning hard.